File: app/src/main/java/com/hypnoweb/hypnowebapp/di/modules/DatabaseModule.java
package com.hypnoweb.hypnowebapp.di.modules;

import android.app.Application;
import android.arch.persistence.room.Room;

import com.hypnoweb.hypnowebapp.database.AppDatabase;
import com.hypnoweb.hypnowebapp.database.movie.MoviesDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by stefan on 12/13/2017.
 */

@Module
public class DatabaseModule {

    public DatabaseModule() {
    }

    @Provides
    @Singleton
    public AppDatabase getDataBase(Application context) {
        return Room.databaseBuilder(context, AppDatabase.class, "HypnoWebDataBase").build();
    }

    @Provides
    public MoviesDao getCatalogDao(AppDatabase appDatabase) {
        return appDatabase.catalogDao();
    }
}

File: app/src/main/java/com/hypnoweb/hypnowebapp/di/modules/DatabaseModule.kt
package com.hypnoweb.hypnowebapp.di.modules

import android.app.Application
import android.arch.persistence.room.Room
import com.hypnoweb.hypnowebapp.database.AppDatabase
import com.hypnoweb.hypnowebapp.database.movie.MoviesDao
import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class DatabaseModule {

    @Provides
    @Singleton
    fun getDataBase(context: Application): AppDatabase {
        return Room.databaseBuilder(context, AppDatabase::class.java, "HypnoWebDataBase").build()
    }

    @Provides
    fun getCatalogDao(appDatabase: AppDatabase): MoviesDao {
        return appDatabase.catalogDao()
    }
}
